sstables/mx/reader.cc

```diff
@@ -1373,22 +1373,35 @@ class mx_sstable_mutation_reader : public mp_row_consumer_reader_mx {
             }
             return make_ready_future();
         }().then([this, pos] {
-            return get_index_reader().advance_to(*pos).then([this] {
-                index_reader& idx = *_index_reader;
-                auto index_position = idx.data_file_positions();
-                if (index_position.start <= _context->position()) {
-                    return make_ready_future<>();
-                }
-                return skip_to(idx.element_kind(), index_position.start).then([this, &idx] {
+            if (_slice.options.contains(query::partition_slice::option::reversed)) {
+                return get_index_reader().advance_reverse(*pos).then([this] {
+                    // _context->reset(sstables::indexable_element::cell);
                     _sst->get_stats().on_partition_seek();
-                    auto open_end_marker = idx.end_open_marker();
+                    auto open_end_marker = _index_reader->end_open_marker();
                     if (open_end_marker) {
                         _consumer.set_range_tombstone(open_end_marker->tomb);
                     } else {
                         _consumer.set_range_tombstone({});
                     }
```

Comment:
I was kind of leaving the handling of the end_open_marker for later, but I just noticed something.
Is it possible that we don't need to change anything concerning the end_open_marker?
After the changes we need something like a start_open_marker that we would get from the promoted_index_block following our current range end (or start if we're talking about reversed ranges). But if I understand correctly, the start_open_marker from the next block would be the same tombstone, as the end_open_marker from the previous one, so because we advance the index to the range end, get the tombstone that we want.